wasm_runtime/src/execution_ctx.rs

```diff
@@ -207,6 +197,51 @@ impl WasmExecutionCtx {
 
         stdout_buf.clone()
     }
+
+    // Helper function to insert a Wasm execution context from the HashMap
+    //  
+    // It checks for duplicated `executionctx_id`.
+    // Returns Result<String, String>, with the inserted Wasm execution context id,
+    // otherswise, in case of error the String will contain the reason.
+    //
+    fn try_insert(wasm_executionctx: WasmExecutionCtx) -> Result<String, String> {
+        let executionctx_id = wasm_executionctx.id.clone();
```

Comment:
I was thinking: why do we need to clone `id`? It's a String, but constructed as a random_hex_id.

For a future improvement: I would change `id` to be a primitive number unsigned type, so that it's `Copy`, and we don't need to clone here and in other places.

Reply:
That's actually great feedback!
Originally, when I started implementing the Wasm executions contexts, I was trying to emulate a Git commit ID, so it was easier to debug.

I think the 'Hex layout' is not really needed anymore and we can use a number instead the String types. I've created #42 to address this 👍